nav2_controller/package.xml

```diff
@@ -19,6 +19,10 @@
   <depend>nav_2d_msgs</depend>
   <depend>nav2_core</depend>
   <depend>pluginlib</depend>
+  <depend>lifecycle_msgs</depend>
```

Comment:
How is bloom missing these on iron and passing the build?

Reply:
Yeah, it's a good question.

Most likely, these are being pulled in by something else.  In this case, it looks like `nav2_util` depends on `lifecycle_msgs`, and so `nav2_controller` inherits it from there.  But since this package directly depends on it, we should technically have this dependency here as well.

Reply:
Ah ok. I thought bloom was supposed to catch transitive dependencies. I must be remembering wrong. Good catch!

Reply:
> Ah ok. I thought bloom was supposed to catch transitive dependencies. I must be remembering wrong. Good catch!

It's complicated.  In point of fact, `bloom` won't check at all that your dependencies are "correct", it will only check that your dependencies can be resolved into system or ROS packages.

The buildfarm goes further, in that it when it goes to build your package, it will start with an absolutely minimal Ubuntu container, and just `apt-get install` exactly what you put in your package.xml.  If a dependency is missing from direct and indirect dependencies, it will definitely find that.  But if you happen to pull in a package via an indirect dependency, things will just work.  That's what is happening here.